### PR TITLE
Add table azure_lb_rule. Close #188

### LINKED ISSUE #188: Add table azure_lb_rule
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.

### PR DESCRIPTION
**Pull request by c0d3r-arnab**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```

arnab@turbotindias-MacBook-Pro azure-test % ./tint.js azure_lb_rule
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/azure_lb_rule []

PRETEST: tests/azure_lb_rule

TEST: tests/azure_lb_rule
Running terraform
azurerm_resource_group.named_test_resource: Creating...
azurerm_resource_group.named_test_resource: Creation complete after 3s [id=/subscriptions/f2ecd657-84a7-4445-9b2a-280aa402eb9f/resourceGroups/turbottest37114]
azurerm_public_ip.named_test_resource: Creating...
azurerm_public_ip.named_test_resource: Still creating... [10s elapsed]
azurerm_public_ip.named_test_resource: Creation complete after 12s [id=/subscriptions/f2ecd657-84a7-4445-9b2a-280aa402eb9f/resourceGroups/turbottest37114/providers/Microsoft.Network/publicIPAddresses/turbottest37114]
azurerm_lb.named_test_resource: Creating...
azurerm_lb.named_test_resource: Creation complete after 9s [id=/subscriptions/f2ecd657-84a7-4445-9b2a-280aa402eb9f/resourceGroups/turbottest37114/providers/Microsoft.Network/loadBalancers/turbottest37114]
azurerm_lb_rule.named_test_resource: Creating...
azurerm_lb_rule.named_test_resource: Creation complete after 6s [id=/subscriptions/f2ecd657-84a7-4445-9b2a-280aa402eb9f/resourceGroups/turbottest37114/providers/Microsoft.Network/loadBalancers/turbottest37114/loadBalancingRules/turbottest37114]

Warning: Version constraints inside provider configuration blocks are deprecated

  on variables.tf line 21, in provider "azurerm":
  21:   version         = "=1.36.0"

Terraform 0.13 and earlier allowed provider version constraints inside the
provider configuration block, but that is now deprecated and will be removed
in a future version of Terraform. To silence this warning, move the provider
version constraint into the required_providers block.


Apply complete! Resources: 4 added, 0 changed, 0 destroyed.

Outputs:

resource_aka = "azure:///subscriptions/f2ecd657-84a7-4445-9b2a-280aa402eb9f/resourceGroups/turbottest37114/providers/Microsoft.Network/loadBalancers/turbottest37114/loadBalancingRules/turbottest37114"
resource_aka_lower = "azure:///subscriptions/f2ecd657-84a7-4445-9b2a-280aa402eb9f/resourcegroups/turbottest37114/providers/microsoft.network/loadbalancers/turbottest37114/loadbalancingrules/turbottest37114"
resource_id = "/subscriptions/f2ecd657-84a7-4445-9b2a-280aa402eb9f/resourceGroups/turbottest37114/providers/Microsoft.Network/loadBalancers/turbottest37114/loadBalancingRules/turbottest37114"
resource_name = "turbottest37114"
subscription_id = "f2ecd657-84a7-4445-9b2a-280aa402eb9f"

Running SQL query: test-get-query.sql
[
  {
    "id": "/subscriptions/f2ecd657-84a7-4445-9b2a-280aa402eb9f/resourceGroups/turbottest37114/providers/Microsoft.Network/loadBalancers/turbottest37114/loadBalancingRules/turbottest37114",
    "name": "turbottest37114",
    "resource_group": "turbottest37114",
    "subscription_id": "f2ecd657-84a7-4445-9b2a-280aa402eb9f",
    "type": "Microsoft.Network/loadBalancers/loadBalancingRules"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "id": "/subscriptions/f2ecd657-84a7-4445-9b2a-280aa402eb9f/resourceGroups/turbottest37114/providers/Microsoft.Network/loadBalancers/turbottest37114/loadBalancingRules/turbottest37114",
    "name": "turbottest37114"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "azure:///subscriptions/f2ecd657-84a7-4445-9b2a-280aa402eb9f/resourceGroups/turbottest37114/providers/Microsoft.Network/loadBalancers/turbottest37114/loadBalancingRules/turbottest37114",
      "azure:///subscriptions/f2ecd657-84a7-4445-9b2a-280aa402eb9f/resourcegroups/turbottest37114/providers/microsoft.network/loadbalancers/turbottest37114/loadbalancingrules/turbottest37114"
    ],
    "name": "turbottest37114",
    "title": "turbottest37114"
  }
]
✔ PASSED

POSTTEST: tests/azure_lb_rule

TEARDOWN: tests/azure_lb_rule

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  id,
  name,
  type,
  provisioning_state,
  etag
from
  azure_lb_rule;
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------+--------------------
| id                                                                                                                                                                   | name          | type               
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------+--------------------
| /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/demo-asr/providers/Microsoft.Network/loadBalancers/arnab_test_lb/loadBalancingRules/arnab_lb_rule | arnab_lb_rule | Microsoft.Network/l
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------+--------------------
> select
  id,
  name,
  provisioning_state
from
  azure_lb_rule
where
  provisioning_state = 'Succeeded';
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------+--------------------
| id                                                                                                                                                                   | name          | provisioning_state 
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------+--------------------
| /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/demo-asr/providers/Microsoft.Network/loadBalancers/arnab_test_lb/loadBalancingRules/arnab_lb_rule | arnab_lb_rule | Succeeded          
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------+--------------------
> select
  id,
  name,
  idle_timeout_in_minutes
from
  azure_lb_rule
order by 
  idle_timeout_in_minutes;
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------+--------------------
| id                                                                                                                                                                   | name          | idle_timeout_in_min
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------+--------------------
| /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/demo-asr/providers/Microsoft.Network/loadBalancers/arnab_test_lb/loadBalancingRules/arnab_lb_rule | arnab_lb_rule | 4                  
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------+--------------------
> 
```
</details>

### CODE REVIEW

File: azure/table_azure_lb_rule.go [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+	resourceGroup := strings.Split(*loadBalancer.ID, "/")[4]
+
+	listLoadBalancerRulesClient := network.NewLoadBalancerLoadBalancingRulesClient(subscriptionID)
+	listLoadBalancerRulesClient.Authorizer = session.Authorizer
+
+	result, err := listLoadBalancerRulesClient.List(ctx, resourceGroup, *loadBalancer.Name)
+	if err != nil {
+		return nil, err
+	}
+
```
Comment (bigdatasourav) [inline azure/table_azure_lb_rule.go:196]:
```suggestion
```

File: azure/table_azure_lb_rule.go [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+	return nil, nil
+}
+
+//// TRANSFORM FUNCTION
+
+func extractLoadBalancerNameFromRuleID(ctx context.Context, d *transform.TransformData) (interface{}, error) {
+	data := d.HydrateItem.(network.LoadBalancingRule)
+	vaultName := strings.Split(*data.ID, "/")[8]
```
Comment (bigdatasourav) [inline azure/table_azure_lb_rule.go:255]:
```suggestion
	loadBalancerName := strings.Split(*data.ID, "/")[8]
```

File: docs/tables/azure_lb_rule.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  name,
+  type,
+  provisioning_state
+from
+  azure_lb_rule
+where
+  provisioning_state = 'Succeeded';
+```
+
+### List load balancer rule order by idle timeout
```
Comment (bigdatasourav) [inline docs/tables/azure_lb_rule.md:34]:
```suggestion
### List load balancer rules order by idle timeout
```

File: azure/table_azure_lb_rule.go [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+)
+
+//// TABLE DEFINITION
+
+func tableAzureLoadBalancerRule(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "azure_lb_rule",
+		Description: "Azure Load Balancer Rule",
+		Get: &plugin.GetConfig{
+			KeyColumns:        plugin.AllColumns([]string{"load_balancer_name", "name", "region", "resource_group"}),
```
Comment (bigdatasourav) [inline azure/table_azure_lb_rule.go:21]:
```suggestion
			KeyColumns:        plugin.AllColumns([]string{"load_balancer_name", "name", "resource_group"}),
```

File: azure/table_azure_lb_rule.go [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+			{
+				Name:        "tags",
```
Comment (bigdatasourav) [inline azure/table_azure_lb_rule.go:146]: tag is not available

File: azure/table_azure_lb_rule.go [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+				Description: ColumnDescriptionAkas,
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("ID").Transform(idToAkas),
+			},
+
+			// Azure standard columns
+			{
+				Name:        "region",
+				Description: ColumnDescriptionRegion,
+				Type:        proto.ColumnType_STRING,
```
Comment (bigdatasourav) [inline azure/table_azure_lb_rule.go:161]: Transform:   transform.FromField("Region").Transform(toLower),

Comment (bigdatasourav) [reply azure/table_azure_lb_rule.go:161]: or you can set it as lower in list call

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Pull request merged**